framework/tlvf/src/src/CmduMessageRx.cpp

```diff
@@ -227,8 +227,12 @@ bool CmduMessageRx::parse(uint8_t *buff, size_t buff_len, bool swap_needed, bool
     
     if (!parse_tlvs)
         return true;
-
-    while (auto tlv = parseNextTlv()) {
+    //need to beter classify message types in order to fit them into dictionary values,
+    //currently -1 is the default TlvMapParser.
+    //for example 4 is for vendor specific messages.
+    int msgtype = -1;
+    CmduParser* parser = (parsers[msgtype]).get();
+    while (std::shared_ptr<BaseClass> tlv = parser->Parse(*this)){
```

Comment:
I'm probably missing something major here, but why do we need a dynamic list of parsers?

The only tricky one is the M1/M2 distinction. But that's actually easy: we can add a single tlvWSC that only has fields type, length and value. Then it's up to the caller to cast it to M1 or M2 as appropriate.